### PR TITLE
Partner Review JS Best Practices

### PR DESCRIPTION
**Pull request by deniodev**
Hi @tumainimaganiko

Kindly review my To-Do List app for JS Best Practices

### CODE REVIEW

File: src/modules/clear.js [hunk 1]
```diff
@@ -0,0 +1,22 @@
+import { task } from './tasks.js';
+
+/* Clear all completed tasks from the todoList array, update their indexes
+and store the updated list in the localStorage. */
+const clearAll = () => {
+  const i = JSON.parse(localStorage.getItem('tasks'));
```
Comment (tumainimaganiko) [inline src/modules/clear.js]: I kindly request you to use descriptive names for your variables

File: src/style.css [hunk 2]
```diff
@@ -0,0 +1,174 @@
...
+.layout {
+  display: flex;
+  flex-direction: column;
+  width: 400px;
+  height: auto;
+  padding: 0;
+  box-shadow: 2px 2px 8px rgba(0, 0, 0, 0.5);
+  background: linear-gradient(145deg, #fff, gray);
+}
+
```
Comment (tumainimaganiko) [inline src/style.css:26]: kindly I recommend you to create a CSS rule for multiple elements to improve the quality of your code

File: src/style.css [hunk 3]
```diff
@@ -0,0 +1,174 @@
+@import url('https://fonts.googleapis.com/css2?family=Roboto:ital,wght@0,100;0,300;0,400;1,100&display=swap');
+
+body {
+  padding: 0;
+  margin: 0;
+  font-family: 'Roboto', sans-serif;
+  background: linear-gradient(145deg, #fff, gray);
+}
+
+#todo-container {
```
Comment (tumainimaganiko) [inline src/style.css]: [Optional]  Specificity: The use of IDs like #todo-container can lead to specificity issues and make the code harder to maintain. Using classes instead would be a better practice.

File: src/style.css [hunk 4]
```diff
@@ -0,0 +1,174 @@
...
+  border: none;
+  left: 3em;
+  line-height: 36px;
+  font-weight: 500;
+  font-size: 16px;
+  background: inherit;
+  font-family: 'Roboto', Courier, monospace;
+}
+
+#todo-form {
```
Comment (tumainimaganiko) [inline src/style.css]: [Optional]  Specificity: The use of IDs like #todo-form can lead to specificity issues and make the code harder to maintain. Using classes instead would be a better practice.

**Review by tumainimaganiko — CHANGES_REQUESTED** [review]
Hi @deniodev ,

You have done a great job so far! 💪 👏
You still need to work on a few issues before moving on to the next project, but you're almost there!

Top highlight 🎯
The Pull Request is following the GitHub Flow ✅
The PR title and description are professional and descriptive ✅
The README is well documented ✅

Required Changes ♻️
image

Optional suggestions
Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better.

You can also consider:

N/A

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise.

**Comment by deniodev** [comment]
Hi @tumainimaganiko 

Kindly see below the changes I made:

- [ ] Add descriptive names to the variables;
- [ ] Add a CSS rule for multiple elements;
- [ ] Use classes instead of ID.

**Review by tumainimaganiko — APPROVED** [review]
Approved 🟢
Hi @tumainimaganiko,

Your project is complete! There is nothing else to say other than... it's time to merge it :shipit:
Congratulations! 🎉

image

Highlights
Great job implementing all changes request by previous reviwers 👏 👏 👏
Professional documented project 📖
Optional suggestions
Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application.

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me (@Trast00) in your question so I can receive the notification.

**Pull request merged**